src/main/java/net/onelitefeather/bettergopaint/listeners/InteractListener.java

```diff
@@ -18,97 +18,100 @@
-import net.onelitefeather.bettergopaint.objects.player.PlayerBrush;
-import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.bukkit.FluidCollisionMode;
 import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
 import org.bukkit.event.block.Action;
 import org.bukkit.event.player.PlayerInteractEvent;
-import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.util.Objects;
 
 public final class InteractListener implements Listener {
 
-    public BetterGoPaint plugin;
+    private final BetterGoPaint plugin;
 
-    public InteractListener(BetterGoPaint main) {
-        plugin = main;
+    public InteractListener(BetterGoPaint plugin) {
+        this.plugin = plugin;
     }
 
-    @SuppressWarnings("deprecation")
     @EventHandler(priority = EventPriority.LOWEST)
-    public void onClick(PlayerInteractEvent e) {
-        if (XMaterial.supports(19)) {
-            if (e.getHand() == EquipmentSlot.OFF_HAND) {
-                return;
-            }
+    public void onClick(PlayerInteractEvent event) {
+        Player player = event.getPlayer();
+
+        if (!player.hasPermission("bettergopaint.use")) {
+            return;
         }
-        if (!e.getPlayer().hasPermission("bettergopaint.use")) {
+
+        ItemStack item = event.getItem();
+        if (item == null) {
             return;
         }
-        if ((e.getAction().equals(Action.RIGHT_CLICK_AIR) || e.getAction().equals(Action.RIGHT_CLICK_BLOCK))) {
-            if (e.getPlayer().getItemInHand().hasItemMeta() && e.getPlayer().getItemInHand().getItemMeta().hasDisplayName() && e
-                    .getPlayer()
-                    .getItemInHand()
-                    .getItemMeta()
-                    .getDisplayName()
-                    .startsWith(" <aqua>♦ ") && e.getPlayer().getItemInHand().getItemMeta().hasLore()) {
-                final ExportedPlayerBrush epb = new ExportedPlayerBrush(e
-                        .getPlayer()
-                        .getItemInHand()
-                        .getItemMeta()
-                        .getDisplayName(), e.getPlayer().getItemInHand().getItemMeta().getLore());
-                final Player p = e.getPlayer();
-                final Location loc;
-                if (e.getAction().equals(Action.RIGHT_CLICK_AIR)) {
-                    loc = p.getTargetBlock(null, 250).getLocation().clone();
-                } else {
-                    loc = e.getClickedBlock().getLocation().clone();
-                }
-                BukkitAdapter.adapt(p).runAsyncIfFree(() ->epb.getBrush().paint(loc, p, epb));
-            }
+
+        if (event.getAction().isLeftClick() && item.getType().equals(Material.FEATHER)) {
```

Comment:
This thing needs to discuss to allow used to use different Items instead of a feather

Reply:
Totally agree, I would even allow any material not just items

Reply:
Any material can be worse and break much stuff